crypto/kdf/pbkdf2.c

```diff
@@ -105,11 +117,14 @@ static int kdf_pbkdf2_ctrl(EVP_KDF_IMPL *impl, int cmd, va_list args)
     case EVP_KDF_CTRL_SET_SALT:
         p = va_arg(args, const unsigned char *);
         len = va_arg(args, size_t);
+        if (impl->range_checks != 0 && len < KDF_PBKDF2_MIN_SALT_LEN)
+            return 0;
         return pbkdf2_set_membuf(&impl->salt, &impl->salt_len, p, len);
 
     case EVP_KDF_CTRL_SET_ITER:
         iter = va_arg(args, int);
-        if (iter < 1)
+        min_iter = impl->range_checks != 0 ? KDF_PBKDF2_MIN_ITERATIONS : 1;
+        if (iter < min_iter)
```

Comment:
Should we be adding errors to the queue in these failure scenarios?

Reply:
Added